src/test/java/tdl/runner/Steps.java

```diff
@@ -0,0 +1,225 @@
+        String username = "tdl-test-cnodejs01";
+        userCommandLineArgs = new String[]{};
+
+        ClientAction clientAction = new ClientAction() {
+            @Override
+            public void afterResponse(RemoteBroker remoteBroker, Request request, Response response) throws BrokerCommunicationException {
+
+            }
+
+            @Override
+            public Optional<Request> getNextRequest(RemoteBroker t) throws BrokerCommunicationException {
+                return Optional.empty();
+            }
+
+            @Override
+            public String getAuditText() {
+                return null;
+            }
+        };
+        UserImplementation userImplementation = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                return null;
+            }
+        };
+        UserImplementation sum = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                sumHit = true;
+                return null;
+            }
+        };
+        UserImplementation hello = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                helloHit = true;
+                return null;
+            }
+        };
+        UserImplementation fizzBuzz = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                fizzBuzzHit = true;
+                return null;
+            }
+        };
+        UserImplementation checkout = new UserImplementation() {
+            @Override
+            public Object process(String... params) {
+                checkoutHit = true;
+                return null;
+            }
+        };
+        Consumer<String> notifyRecordSystemCallback = new Consumer<String>() {
+            @Override
+            public void accept(String s) {
+                recordingSystemCallbackText = s;
+            }
+        };
+
+        // start in a background thread, check if thread finished for exit.
+
+        ClientRunner clientRunner = ClientRunner.forUsername(username)
+                .withServerHostname(hostname)
+                .withPort(port)
+                .withJourneyId(journeyId)
+                .withColours(true)
+                .withDeployCallback(() -> deployCallbackHit = true)
+                .withDeployAction(clientAction)
+                .withRecordingSystemOk(true)
+                .withBufferedReader(bufferedReader)
+                .withOutputStream(printStream)
+                .withNotifyRecordSystemCallback(notifyRecordSystemCallback)
+                .withSolutionFor("sum", p -> sum)
+                .withSolutionFor("hello", p -> hello)
+                .withSolutionFor("fizz_buzz", p -> fizzBuzz)
+                .withSolutionFor("checkout", p -> checkout);
+
+        clientRunnerThread = new Thread(() -> clientRunner.start(userCommandLineArgs));
```

Comment:
Why is this running on a Thread?

Reply:
Won't it otherwise block when waiting for input?